setup.py

```diff
@@ -52,6 +52,7 @@ def run(self):
 
 
 install_requires = [
+    "python-dateutil<2.8.1,>=2.1",  # Compatibility with botocore
```

Comment:
Hm, never seen this one before. What is conflicting with it? We are not using it, is it some other dep?

Reply:
yep, some other dependency: `botocore`, part of `awscli`.

Reply:
Yeah, but botocore already has those requirements set https://github.com/boto/botocore/blob/develop/setup.py#L43 , I just don't quite understand why we need to list them here one more time?

Reply:
`azure-storage-blob==2.1.0` is using: `python-dateutil [required: Any, installed: 2.8.0]`

I imagine it downloads the latest version and then it collides. Let me test it very quickly, maybe it is a leftover, @efiop

Reply:
Yep:
```
ERROR: botocore 1.13.33 has requirement python-dateutil<2.8.1,>=2.1; python_version >= "2.7", but you'll have python-dateutil 2.8.1 which is incompatible.
```

Azure is installing first the latest version and then boto collides with it

Reply:
Yeah, this happens because PIP doesn't have a full dependency solver, which we will get in the near future (fingers crossed)

Reply:
Yeah, hoping for the Santa to bring a new dependency resolver for pip 🙏 😄 

@mroutis Interesting, not seeing dateutil in https://github.com/Azure/azure-storage-python/blob/master/azure-storage-blob/setup.py . 🤔

Reply:
@efiop , is under `azure-storage-common`

Reply:
https://github.com/Azure/azure-storage-python/blob/4e03ff8c9d6c76a38666a8ceb8937593e93fb7eb/azure-storage-common/setup.py#L76

Reply:
for the record, I used `pipdeptree` to detect this.

Reply:
@mroutis Ah, got it. Thanks a lot for the research!